### ISSUE TITLE
`appearance -> theme` validation warning when starting napari

### ISSUE THREAD

**Issue by lukasz-migas** [napari/napari#3270]
## 🐛 Bug

PR #3034 introduced hook spec to add custom themes which are dynamically loaded at app startup. 
If the last active theme is not one of the defaults (dark/light) and is provided by the hook spec and napari is closed, the last used value is exported to settings file. Next time napari is started, that theme is not loaded yet so a warning is emitted and napari reverts back to safe default.

## To Reproduce

(this is not a great example because the theme provided by `new_theme` example is not provided by the hook spec)

1. Run the `examples/new_theme.py` example.
2. Close the window.
3. Re-run the example. Warning will be shown.

![image](https://user-images.githubusercontent.com/25161821/131193061-9e8b98c9-bcab-468e-b737-18a55e72f486.png)

## Expected behavior

The validator should take into account non-default themes into account and not emit this warning.
The list of themes provided by `available_themes` should populated before settings are validated.

**Comment by goanpeca** [comment]
Settings are parsed early on, so I am not sure about this. Thoughts @tlambert03 ?

**Comment by lukasz-migas** [comment]
Indeed,  I couldn't think of how to solve this to be honest. Perhaps once the new plugin architecture is developed/finalised it will be possible to fix? That of course will only fix the plugin-provided themes issue.

**Comment by tlambert03** [comment]
yeah, definitely easier with the manifests... probably possible to fix now too? but would need to look deeper (and it might not be an elegant fix)

**Comment by psobolewskiPhD** [comment]
I think this is fixed with:
https://github.com/napari/napari/pull/5150
https://github.com/napari/napari/pull/5605
@czaki @brisvag y'all prolly have the most insight here, but I don't get any errors or warnings:
```
╰─ python examples/new_theme.py                                    (napari-dev) ─╯
Originally themes ['dark', 'light', 'system']
New themes ['dark', 'light', 'blue', 'system']
╭─    ~/dev/napari   main *1 ·································· ✔   6s ─
╰─ python examples/new_theme.py                                    (napari-dev) ─╯
Originally themes ['dark', 'light', 'system']
New themes ['dark', 'light', 'blue', 'system']
```

**Comment by Czaki** [comment]
It was fixed in #3270
